File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) 
    {
        int min=0;
        int max=0;
        int n=nums.length;
        for(int i=0;i<n;i++)
        {
            if(nums[i]<nums[min])
            {
                min=i;
            }
            if(nums[i]>nums[max])
            {
                max=i;
            }
        }

        int left=Math.max(min,max)+1;
        int right=n-Math.min(min,max);
        int both=Math.min(min,max)+1 + n-Math.max(min,max);

        return Math.min(left,Math.min(right,both));

    }
}

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.py
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        mini=0
        maxi=0
        for i in range(len(nums)):
            if nums[i]>nums[maxi]:
                maxi=i
            if nums[i]<nums[mini]:
                mini=i
        n=len(nums)
        left=max(mini,maxi)+1
        right=n-min(mini,maxi)
        both=min(mini,maxi)+1 + n-max(mini,maxi)
        return min(left,min(right,both))
        